### DISCUSSION TITLE
[BUG] Add-PnPHubSiteAssociation randomly fails with inexplicable "File not found"

### DISCUSSION THREAD

**Discussion by iloving** [pnp/powershell#2299] [General]
I have a script that creates a sharepoint site and then adds it to our hub site.  One of the last lines in my script is:

`Add-PnPHubSiteAssociation -Connection $Connection -Site "$SiteFullURL" -HubSite "$HubSite"`

### Expected behavior 
I expect it to work.

### Actual behavior 
Returns the error:
Add-PnPHubSiteAssociation: /$MYPATH/MasterSiteCreateScript.ps1:379:5
Line |
 379 |      Add-PnPHubSiteAssociation -Connection $Connection -Site "$SiteFul …
     |      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | File Not Found.

### Steps to reproduce behavior
I have no idea.  It normally works, until it doesn't.  It's completely unpredictable.

### What is the version of the Cmdlet module you are running?
1.11.0

### Which operating system/environment are you running PnP PowerShell on?
- [ ] Windows
- [ ] Linux
- [X] MacOS
- [ ] Azure Cloud Shell
- [ ] Azure Functions
- [ ] Other : please specify

**Comment by CallumCrowley** [comment]
Can you set tracing to debug using Set-PnPTraceLog?

That should give you more information which you can provide here so that we can better understand the problem.

**Comment by gautamdsheth** [comment]
Moving this to discussions since this a random issue with the environment itself and not the cmdlet.
